src/credential/mod.rs

```diff
@@ -1,45 +1,159 @@
-use std::sync::Arc;
+// use std::sync::Arc;
 
-use serde::{Deserialize, Serialize};
+// use serde::{Deserialize, Serialize};
 
-use json_vc::{JsonVc, JsonVcEncodingError, JsonVcInitError};
-use jwt_vc::{JwtVc, JwtVcInitError};
-use mdoc::{Mdoc, MdocEncodingError, MdocInitError};
+// use json_vc::{JsonVc, JsonVcEncodingError, JsonVcInitError};
+// use jwt_vc::{JwtVc, JwtVcInitError};
+// use mdoc::{Mdoc, MdocEncodingError, MdocInitError};
 
-use crate::{CredentialType, KeyAlias, Uuid};
+// use crate::{CredentialType, KeyAlias, Uuid};
 
 pub mod json_vc;
 pub mod jwt_vc;
 pub mod mdoc;
-pub mod sd_jwt_vc;
+pub mod sd_jwt;
+
+use std::sync::Arc;
+
+use crate::{CredentialType, KeyAlias, Uuid};
+use json_vc::{JsonVc, JsonVcInitError};
+use jwt_vc::{JwtVc, JwtVcInitError};
+use mdoc::{Mdoc, MdocEncodingError, MdocInitError};
+use oid4vp::core::{
+    credential_format::ClaimFormatDesignation, presentation_definition::PresentationDefinition,
+};
+use sd_jwt::{SdJwt, SdJwtError};
+use serde::{Deserialize, Serialize};
+use ssi::prelude::AnyJsonCredential;
 
 /// An unparsed credential, retrieved from storage.
-#[derive(Debug, Serialize, Deserialize, uniffi::Record)]
+#[derive(Debug, Clone, Serialize, Deserialize, uniffi::Object)]
```

Comment:
IIRC, this was due to an error when trying to provide an exported interface (`#[uniffi::export]`), which I added a constructor method for the `Credential` type. That said, we may not expect the `Credential` to have a constructor directly, in which case we may be able to remove the exported interface and add back the `Record`.

I've add accessor methods for the fields, and updated in the kotlin/swift code to use the `type()` accessor method.

While the record is nice to access the fields directly, I think we should prefer to use an `Object` type here. Was there a reason against using an `Object` in favor of a `Record`?

It seems like we might want a constructor method that returns the `Arc<Credential>` reference for usability. I don't see a significant downside to using `Object` over `Record`, but I welcome your thoughts on this.

Reply:
Since the construction of the Credential is really only happening form the VDC collection, I don't think the constructor is needed, and therefore we can revert to using the `Record`. I've reverted this in the latest commit to use the `Record` type.